File: 066 Plus One/plus_one.cs
/*	
 *	LeetCode Submissions by Xinyu Chen
 *	Plus One
 *	https://leetcode.com/problems/plus-one/
 *	Runtime: 524 ms
 */
 
public class Solution {
    public int[] PlusOne(int[] digits) {
        int carry = 0;
		List<int> res = new List<int>();
		if (digits[digits.Length - 1] == 9)
		{
		    res.Add(0);
		    carry = 1;
		}
		else
		{
		    res.Add(digits[digits.Length - 1] + 1);
		}
		for (int i = digits.Length - 2; i >= 0; i--)
		{
		    if (digits[i] == 9 && carry == 1)
		    {
		        res.Add(0);
		    }
		    else
		    {
		        res.Add(digits[i] + carry);
		        carry = 0;
		    }
		}
		if (carry == 1)
		{
		    res.Add(1);
		}
		res.Reverse();
		return res.ToArray();
    }
}

File: 066 Plus One/plus_one.java
/*	
 *	LeetCode Submissions by Xinyu Chen
 *	Plus One
 *	https://leetcode.com/problems/plus-one/
 *	Runtime: 321 ms
 */
 
public class Solution {
    public int[] plusOne(int[] digits) {
        int carry = 0;
        ArrayList<Integer> res = new ArrayList<Integer>();
        if (digits[digits.length - 1] == 9) {
            res.add(0);
            carry = 1;
        } else {
            res.add(digits[digits.length - 1] + 1);
        }
        for (int i = digits.length - 2; i >= 0; i--) {
            if (digits[i] == 9 && carry == 1) {
                res.add(0);
            } else {
                res.add(digits[i] + carry);
                carry = 0;
            }
        }
        if (carry == 1) {
            res.add(1);
        }
        Collections.reverse(res);
        int[] ret = new int[res.size()];
        for (int i = 0; i < res.size(); i++) ret[i] = res.get(i).intValue();
        return ret;
    }
}